src/duplicacy_snapshotmanager.go

```diff
@@ -2227,6 +2231,9 @@ func (manager *SnapshotManager) DownloadFile(path string, derivationKey string)
 
 // UploadFile uploads a non-chunk file from the storage.
 func (manager *SnapshotManager) UploadFile(path string, derivationKey string, content []byte) bool {
+    if manager.config.dryRun {
```

Comment:
I think this check isn't necessary.  UploadFile is called by UploadSnapshot, and there is already a check there.